pages/DetailsTrip/[city].js

```diff
@@ -0,0 +1,137 @@
+import { useRouter } from "next/router";
+import {
+  StyleButton,
+  StyleList,
+  StyleDiv,
+  StyleCountry,
+  StyleSection,
+  StyleSpan,
+  StyleH4,
+  StyleTotalPrice,
+} from "../../src/components/StyledTripDetails";
+import Header from "@/src/components/Header/Header";
+import { countries } from "@/lib/db";
+
+export default function DetailsOfTrip() {
+  const router = useRouter();
+  const cityName = router.query.city;
+
+  /* Get details about the city*/
+  const country = countries.find((country) =>
+    country.cities.find((city) => city.name === cityName)
+  );
+
+  const city = country?.cities.find((city) => city.name === cityName);
+
+  console.log(city);
+
+  if (!city) {
+    return <h2>City not found</h2>;
+  }
+
+  /* Calculate Hotel Prices */
+  const totalHotelPrice = city.hotel.map((hotel) => hotel.price);
+  console.log(totalHotelPrice);
+  const totalPriceOfHotels = totalHotelPrice.reduce(
+    (hotelPrice, price) => hotelPrice + Number(price),
+    0
+  );
+  console.log(totalPriceOfHotels);
+
+  /* Calculate Place Prices */
+  const totalPlacePrice = city.place.map((place) => place.price);
+  console.log(totalPlacePrice);
+  const totalPriceOfPlaces = totalPlacePrice.reduce(
+    (placePrice, price) => placePrice + Number(price),
+    0
+  );
+  console.log(totalPriceOfPlaces);
+
+  /* Calculate food Prices */
+  const totalFoodPrice = city.food.map((food) => food.price);
+  console.log(totalFoodPrice);
+  const totalPriceeOfFoods = totalFoodPrice.reduce(
+    (foodPrice, price) => foodPrice + Number(price),
+    0
+  );
+  console.log(totalPriceeOfFoods);
+
+  let totalPrice = totalPriceOfHotels + totalPriceOfPlaces + totalPriceeOfFoods;
+  console.log(totalPrice);
+
+  /* Back to thr home page*/
+  const backHome = () => {
+    router.push("/");
+  };
+
+  /* Return details about the City */
+  return (
+    <>
+      <Header title="Details of traveling" />
+      <StyleButton onClick={backHome}> Home</StyleButton>
+      <StyleSection>
+        <StyleCountry>{country.name}</StyleCountry>
+
+        <StyleDiv>
+          <StyleH4>{city.name}</StyleH4>
+          <StyleSpan>
+            {city.startDate} - {city.endDate}
+          </StyleSpan>
+        </StyleDiv>
+        <StyleH4>Hotels:</StyleH4>
+        <ul>
+          {city.hotel.map((hotel) => (
+            <StyleList key={hotel.name}>
```

Comment:
Make sure that your key stays unique. If you fetch for hotel names you gonna run into issues if a hotel has the same name. Either use UID library or try to plan out unique IDs in your db object.

Reply:
Hi, Josh
Thank you very much for your review and all advices